components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/scehduler/CoordinatedTaskScheduler.java

```diff
@@ -172,21 +173,24 @@ private void scheduleAssignedTasks(CoordinatedTask.States state) throws TaskCoor
             return;
         }
         List<String> deployedCoordinatedTasks = taskManager.getAllCoordinatedTasksDeployed();
-        tasksOfThisNode.forEach(taskName -> {
+        for (String taskName : tasksOfThisNode) {
             if (deployedCoordinatedTasks.contains(taskName)) {
                 if (LOG.isDebugEnabled()) {
                     LOG.debug("Submitting retrieved task [" + taskName + "] to the task manager.");
                 }
                 try {
                     taskManager.scheduleCoordinatedTask(taskName);
-                } catch (TaskException e) {
-                    LOG.error("Exception occurred while scheduling coordinated task : " + taskName, e);
+                } catch (TaskException ex) {
+                    if (!TaskException.Code.DATABASE_ERROR.equals(ex.getCode())) {
+                        taskStore.updateTaskState(Collections.singletonList(taskName), CoordinatedTask.States.NONE);
```

Comment:
What if we add these to a failed list and perform this at the end of the loop, will reduce the time for the db callse? Also, don't we need to unassign the task in this case?

Reply:
Or do we need a logic like, if scheduling a task fails 2/3 times consecutively, we will unassign it?

Reply:
+1 will move to end of loop. No point in un-assigning the task , If the node leaves the cluster, it will get assigned to something else automatically . If not , even if we un assign  it might get assigned to the same node again.